go/vt/vttablet/tabletmanager/vreplication/engine_test.go

```diff
@@ -426,7 +426,8 @@ func TestWaitForPosError(t *testing.T) {
 
 	dbClient.ExpectRequest("select pos, state, message from _vt.vreplication where id=1", &sqltypes.Result{Rows: [][]sqltypes.Value{{}}}, nil)
 	err = vre.WaitForPos(context.Background(), 1, "MariaDB/0-1-1084")
-	want = "unexpected result: &{[] 0 0 [[]]  0 }"
+	want = fmt.Sprintf("unexpected result: %v", &sqltypes.Result{Rows: [][]sqltypes.Value{{}}})
```

Comment:
@rafer I think we should actually fix the error message generated here. I don't think it's very clear what this is in the first place. The error message does not seem very user friendly if it contains `&{[] 0 0 [[]]  0 }`. I don't think that's meaningful in any way for a Vitess user? 

Something along these lines maybe, so instead of:

```go
		case len(qr.Rows) > 1 || len(qr.Rows[0]) != 3:
			return fmt.Errorf("unexpected result: %v", qr)
```

We do something like:

```go
		case len(qr.Rows) > 1:
			return fmt.Errorf("more rows received than expected, got %v instead of 1", len(qr.Rows))
		case len(qr.Rows[0]) != 3:
			return fmt.Errorf("more columns received than expected, got %v instead of 3", len(qr.Rows[0]))
```

Reply:
Alternatively, just printing `qr.Rows` instead of the whole `qr` already makes it look a lot better as well, since rows have a proper `.String()` implementation for debugging.

Reply:
Update 👍